### PR TITLE
[Trusted Types] Fixing violations sourced from setAttribute wrapper function

### PR DESCRIPTION
**Pull request by youssef-attia**
## What is the purpose of this pull request?

PR to fix violations coming from setAttribute wrapper function to help with making CKEditor [Trusted Types](https://developer.mozilla.org/en-US/docs/Web/API/Trusted_Types_API) compatible.

setAttribute is used as a wrapper function for the [DOM setAttribute function] (https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute). The function acts as a XSS sink since it gives unrestricted access in setting attributes like 'src'. This function is used primarily for script loading so security in this segment is essential.

### CODE REVIEW

File: core/scriptloader.js [hunk 1]
```diff
@@ -114,10 +114,30 @@ CKEDITOR.scriptLoader = ( function() {
...
-					script.setAttributes( {
-						type: 'text/javascript',
-						src: url
-					} );
+					if (self.trustedTypes && self.trustedTypes.createPolicy) {
+						const policy = self.trustedTypes.createPolicy(
+						  'scriptloader#loadScript',
+						  {
+							createScriptURL: function (url) {
+								const regexURL = /^https:\/\/example\.com(\/.*)?$/;
```
Comment (eozmen410) [inline core/scriptloader.js]: is this supposed to be example.com?

Comment (youssef-attia) [reply core/scriptloader.js]: For now, we were planning to keep it as a placeholder example.com and add our internal regex in the build file unless there was some better way to keep a placeholder here?

Comment (youssef-attia) [reply core/scriptloader.js]: Changed this to a legacy conversion

**Comment by youssef-attia** [comment]
image
Tests all pass on rerun

File: core/scriptloader.js [hunk 2]
```diff
@@ -114,10 +114,26 @@ CKEDITOR.scriptLoader = ( function() {
 
 					// Create the <script> element.
 					var script = new CKEDITOR.dom.element( 'script' );
-					script.setAttributes( {
-						type: 'text/javascript',
-						src: url
-					} );
+					if (self.trustedTypes && self.trustedTypes.createPolicy) {
+						// Legacy converted for now because of the possibility of using alternate CDNs for serving scripts.
+						const policy = self.trustedTypes.createPolicy(
```
Comment (eozmen410) [inline core/scriptloader.js]: should this be `var`?

Comment (youssef-attia) [reply core/scriptloader.js]: Yep just fixed

**Review by eozmen410 — APPROVED** [review]

**Pull request merged**